examples/tn_simulation.py

```diff
@@ -0,0 +1,157 @@
+"""
+Usage and application of tensor network simulations
+=====================================================
+
+In this example, we will illustrate the usage and application of TN simulation of MBQC.
+
+Firstly, let's import the relevant modules:
+"""
+
+from functools import reduce
+
+import cotengra as ctg
+import matplotlib.pyplot as plt
+import numpy as np
+import opt_einsum as oe
+from scipy.optimize import minimize
+
+from graphix import Circuit
+
+# This application will be for the QAOA (Quantum Approximate Optimization Algorithm),
```

Comment:
could you check that this file renders well with sphinx-gallery? for example, this comment needs to be 
```python
# %%
# This application ...
```

see for example https://github.com/TeamGraphix/graphix/blob/master/examples/qft.py and https://sphinx-gallery.github.io/stable/syntax.html for how to write rst-style docstring in python file. Render this example locally, by running `make html` in docs folder (check requirements.txt in docs folder so you can run sphinx-gallery.

Reply:
Sure, in the following commits I will pay attention to it